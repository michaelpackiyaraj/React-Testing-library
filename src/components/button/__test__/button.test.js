import React from "react";
import ReactDOM from 'react-dom';
import Button from './../';

import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

it("renders without crasing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div);
})

it("render button correctly", ()=>{
    const { getByTestId } = render(<Button label="click me"></Button>);
    expect(getByTestId('button')).toHaveTextContent("click me");
})

it("render save button correctly", ()=>{
    const { getByTestId } = render(<Button label="save"></Button>);
    expect(getByTestId('button')).toHaveTextContent("save");
})

it("matches snapshot", () => {
    const tree = renderer.create(<Button label="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});