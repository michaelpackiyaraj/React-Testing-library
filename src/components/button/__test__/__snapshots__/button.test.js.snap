// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="my-button"
  data-testid="button"
>
  hello 
  save
</div>
`;
